spark/src/main/java/org/apache/iceberg/spark/PruneColumnsWithReordering.java

```diff
@@ -72,7 +72,7 @@ public Type schema(Schema schema, Supplier<Type> structResult) {
   public Type struct(Types.StructType struct, Iterable<Type> fieldResults) {
     Preconditions.checkNotNull(struct, "Cannot prune null struct. Pruning must start with a schema.");
     Preconditions.checkArgument(current instanceof StructType, "Not a struct: %s", current);
-    StructType s = (StructType) current;
+    StructType requestedStruct = (StructType) current;
```

Comment:
I think `currenStruct` would be a better name since the variable that it is assigned from is `current`.

Reply:
I was inspired by other existing places such as [this](https://github.com/apache/incubator-iceberg/blob/master/spark/src/main/java/org/apache/iceberg/spark/PruneColumnsWithReordering.java#L107). It was a bit unclear what `current` means until I saw that `current = requestedType` initially. Since this class is about pruning `requested` made sense to me.

If we keep `requested` in the name, conditions are self-descriptive `requestedDecimal.scale() == decimal.scale()`.